common/src/main/scala/net/psforever/objects/avatar/PlayerControl.scala

```diff
@@ -760,6 +967,10 @@ object PlayerControl {
     target.Actor ! JammableUnit.ClearJammeredSound()
     target.Actor ! JammableUnit.ClearJammeredStatus()
     events ! AvatarServiceMessage(nameChannel, AvatarAction.Killed(player_guid, target.VehicleSeated)) //align client interface fields with state
+    //implants off
+    target.Actor ! Player.StaminaChanged(-target.Stamina)
+    target.Stamina = 0
```

Comment:
This feels weird, having to set the Stamina, but then also send off a notification that the stamina has changed to `PlayerControl`, and is probably very human error prone in that it's very easy to forget to do both things in other areas of the codebase.

That was my original thinking behind having the the `Stamina` setter also call out to `PlayerControl` by itself, we would never forget to do so.

Food for thought?

Reply:
We have do this for other state changes too.  For things like health updates, for example, I think you're not appreciating how the workflow is set up to occur at the tail end of a damage packet resolution.  If you set a player's health to zero at any time he doesn't just die.  You can send the update packet to immobilize him on the clients; but, to actually kill the player, the `Die` message has to be sent server-side.  Otherwise, the server will determine incorrect actions must be performed during respawn operations.

Reply:
Actually, am I missing something here? Aren't we assinging `player.Stamina` twice in this example?

`Player.StaminaChanged` calls `StaminaChanged` which has the following line:
`val afterStamina = player.Stamina += changeInStamina`

So we're assinging the `changeInStamina` to `player.Stamina`, then assinging the result of that to `afterStamina`, but then also setting `target.Stamina` to `0` right after

Reply:
The player is already dead by accounts of line 965, so they won't regen stamina anymore after that point.  The do, however, need to propagate messages that turn their implants off and uninitialize them.  `StaminaChanged` is one message that can do that and it used to be connected to that mutator.  I'll eliminate the mutator call for `player.Stamina` on line 972 but the client update to 0 stamina on line 993 should remain.